README.md

```diff
@@ -115,86 +116,55 @@ used to incorporate proxies into JavaScript:  That pattern specified...
 
 The static methods first do a `Promise.resolve` on their first
 argument, to coerce it to a promise with these internal methods.  Thus, for
 example,
 
 ```js
-Promise.eventualGet(p, prop)
+Promise.delegated.eventualGet(p, prop)
 ```
 actually does the equivalent of
-```
-Promise.resolve(p).[[EventualGet]](prop)
+```js
+Promise.resolve(p).[[EventualGet]](prop, {})
 ```
 
 Via the internal methods, the static methods cause either the default behavior,
 or, for delegated promises, the behavior that calls the associated handler trap.
 
 | Static Method | Default Behavior | Handler trap |
 | --- | --- | --- |
-| `eventualGet(p, prop)` | `p.then(t => t[prop])` | `h.eventualGet(t, prop)` |
-| `eventualApply(p, args)` | `p.then(t => t(...args))` | `h.eventualApply(t, args)` |
-| `eventualSend(p, prop, args)` | `p.then(t => t[prop](...args))` | `h.eventualSend(t, prop, args)` |
+| `eventualGet(p, prop, opts)` | `p.then(t => t[prop])` | `h.eventualGet(t, prop, opts)` |
+| `eventualApply(p, args, opts)` | `p.then(t => t(...args))` | `h.eventualApply(t, args, opts)` |
+| `eventualSend(p, prop, args, opts)` | `p.then(t => t[prop](...args))` | `h.eventualSend(t, prop, args, opts)` |
 
 To protect against reentrancy, the proxy internal method postpones the
 execution of the handler trap to a later turn, and immediately returns a
 promise for what the trap will return.  For example, the [[EventualGet]] internal
 method of a delegated promise is effectively
 
 ```js
-p.then(t => h.eventualGet(t, prop))
+p.then(t => h.eventualGet(t, prop, opts))
 ```
 
-Sometimes, these operations will be used to cause remote effects while ignoring
-the local promise for the result.  For distributed messaging protocols, the
-extra bookkeeping for these return results are sufficiently expensive that we
-should be able to avoid it when it is not needed.  To support this, we
-introduce the "SendOnly" variants of these methods.
-
-| Internal Method | Static Method |
-| --- | --- |
-| `p.[[EventualGetOnly]](prop)`       | `eventualGetOnly(p, prop)` |
-| `p.[[EventualApplyOnly]](args)`     | `eventualApplyOnly(p, args)` |
-| `p.[[EventualSendOnly]](prop, args)`| `eventualSendOnly(p, prop, args)` |
-
-| Static Method | Handler trap |
-| --- | --- |
-| `eventualGetOnly(p, prop)`        | `h.eventualGetOnly(t, prop)` |
-| `eventualApplyOnly(p, args)`      | `h.eventualApplyOnly(t, args)` |
-| `eventualSendOnly(p, prop, args)` | `h.eventualSendOnly(t, prop, args)` |
-
-
-
-No matter what a \*Only handler trap returns, the proxy internal
-[[\*Only]] method always immediately returns `undefined`.
-
-When an "Only" trap is absent, the trap behavior defaults to the
-corresponding non-Only trap.  But again, the proxy internal [[\*Only]]
-method always immediately returns `undefined`, and so is effectively, for
-example:
-
-```js
-void p.then(t => h.eventualGet(t, prop))
-```
-
-### `E` and `E.sendOnly` convenience proxies
+### `E` convenience proxy maker
```

Comment:
As long as you're keeping E and adding `opt` elsewhere, you may as well include `opt` as an optional second argument to `E`.

On reflection, I think we should propose the basic `E` and `E.get`, because without either that or special syntax, this proposal by itself doesn't create a system anyone would use. Omitting `E.when` for now is fine. That doesn't prevent us from adding it to our shim and proposing it in a follow-on proposal if we can't come up with any better way of expressing this.

`E.when` suggests the notorious `E.where`. The possibility of providing both together gives us an even stronger reason to hold it back until a follow-on proposal.

Reply:
I made an issue for it: #28